core/src/main/scala/org/http4s/headers/Content-Length.scala

```diff
@@ -12,18 +12,24 @@ import org.http4s.util.Writer
   *
   * @param length the length; throws an `IllegalArgumentException` if negative
   */
-final case class `Content-Length`(length: Long) extends Header.Parsed {
-  require(length >= 0L)
-
+sealed abstract case class `Content-Length`(length: Long) extends Header.Parsed {
   override def key: `Content-Length`.type = `Content-Length`
   override def renderValue(writer: Writer): writer.type = writer.append(length)
+  def modify(f: Long => Long): `Content-Length` = new `Content-Length`(f(length)) {}
 }
 
 object `Content-Length` extends HeaderKey.Internal[`Content-Length`] with HeaderKey.Singleton {
+  val zero = new `Content-Length`(0) {}
+
+  def apply(length: Long): ParseResult[`Content-Length`] = fromLong(length)
```

Comment:
I'm rather ambivalent.

I'm trying to decide how to help folks transition, as this is a large break to the API.